find_coordinator_response.go

```diff
@@ -60,10 +60,22 @@ func (f *FindCoordinatorResponse) encode(pe packetEncoder) error {
 		}
 	}
 
+	if f.Coordinator == nil {
+		pe.putInt32(0)
+		if err := pe.putString(""); err != nil {
+			return nil
+		}
+		pe.putInt32(0)
+		if f.Version >= 1 {
```

Comment:
This is not matched by the fact that we hard-code passing version 0 into `Coordinator.encode`?

Reply:
Hm. We are currently sharing the same decode/encode method for `Metadata*` and `FindCoordinator*` but since version 1 of the `MetadataResponse` they are not the equal (`Metadata` has rack whereas `FindCoordinator` does not.)